includes/Classifai/Providers/OpenAI/ChatGPT.php

```diff
@@ -0,0 +1,510 @@
+	 */
+	public function get_provider_debug_information( $settings = null, $configured = null ) {
+		if ( is_null( $settings ) ) {
+			$settings = $this->sanitize_settings( $this->get_settings() );
+		}
+
+		$authenticated  = 1 === intval( $settings['authenticated'] ?? 0 );
+		$enable_excerpt = 1 === intval( $settings['enable_excerpt'] ?? 0 );
+
+		return [
+			__( 'Authenticated', 'classifai' )     => $authenticated ? __( 'yes', 'classifai' ) : __( 'no', 'classifai' ),
+			__( 'Generate excerpt', 'classifai' )  => $enable_excerpt ? __( 'yes', 'classifai' ) : __( 'no', 'classifai' ),
+			__( 'Excerpt length', 'classifai' )    => $settings['length'] ?? 2,
+			__( 'Temperature value', 'classifai' ) => $settings['temperature'] ?? 1,
+			__( 'Latest response', 'classifai' )   => $this->get_formatted_latest_response( 'classifai_openai_chatgpt_latest_response' ),
+		];
+	}
+
+	/**
+	 * Format the result of most recent request.
+	 *
+	 * @param string $transient Transient that holds our data.
+	 * @return string
+	 */
+	private function get_formatted_latest_response( string $transient = '' ) {
+		$data = get_transient( $transient );
+
+		if ( ! $data ) {
+			return __( 'N/A', 'classifai' );
+		}
+
+		if ( is_wp_error( $data ) ) {
+			return $data->get_error_message();
+		}
+
+		return preg_replace( '/,"/', ', "', wp_json_encode( $data ) );
+	}
+
+	/**
+	 * Common entry point for all REST endpoints for this provider.
+	 * This is called by the Service.
+	 *
+	 * @param int    $post_id The Post Id we're processing.
+	 * @param string $route_to_call The route we are processing.
+	 * @return string|WP_Error
+	 */
+	public function rest_endpoint_callback( $post_id = 0, $route_to_call = '' ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'post_id_required', esc_html__( 'A valid post ID is required to generate an excerpt.', 'classifai' ) );
+		}
+
+		$return = '';
+
+		// Handle all of our routes.
+		switch ( $route_to_call ) {
+			case 'excerpt':
+				$return = $this->generate_excerpt( $post_id );
+				break;
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Generate an excerpt using ChatGPT.
+	 *
+	 * @param int $post_id The Post Id we're processing
+	 * @return string|WP_Error
+	 */
+	public function generate_excerpt( int $post_id = 0 ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'post_id_required', esc_html__( 'Post ID is required to generate an excerpt.', 'classifai' ) );
+		}
+
+		$settings = $this->get_settings();
+
+		// These checks (and the one above) happen in the REST permission_callback,
+		// but we run them again here in case this method is called directly.
+		if ( empty( $settings ) || ( isset( $settings['authenticated'] ) && false === $settings['authenticated'] ) || ( isset( $settings['enable_excerpt'] ) && 'no' === $settings['enable_excerpt'] ) ) {
+			return new WP_Error( 'not_enabled', esc_html__( 'Excerpt generation is turned off or authentication failed.', 'classifai' ) );
```

Comment:
Perhaps "Excerpt generation is disable or OpenAI ChatGPT authentication failed." though I guess if we eventually integrate another service for excerpt generation that we wouldn't want to call out OpenAI if something like Azure is actually in use.

Reply:
Updated in [9542a02](https://github.com/10up/classifai/pull/405/commits/9542a02ebf8fd899babb02794e209748dc76903e)